### PR TITLE
[stable/kubernetes-dashboard] #7516 Allow HTTP connection

### LINKED ISSUE #7516: [stable/kubernetes-dashboard] Allow HTTP connection
There are situations where the kubernetes dashboard is hosted behind ingress controllers / proxies that perform TLS termination already. Many ingress controllers / proxies (cloudflare argo ingress, oauth2_proxy) expect proxied services to be HTTP, and do not support HTTPS services. 

Recommend adding an input value to setup deployment to allow HTTP traffic (which is supported by k8s-dashboard binary itself via `--insecure-port`, `--insecure-bind-address` and `--enable-insecure-login` option. 

Reference: [On Securing the Kubernetes Dashboard by Joe Beda @ Heptio
](https://blog.heptio.com/on-securing-the-kubernetes-dashboard-16b09b1b7aca)

### PR DESCRIPTION
**Pull request by mbarrien**
#### What this PR does / why we need it:
This PR allows service Kubernetes dashboard with unencrypted HTTP connections via a values flag. This allows it to be served from behind an ingress controller/proxy that do not support self signed certificates and that handle their own authentication/authorization.

We add a `enableInsecureLogin` flag, defaulting to `false` for backward compatibility/default security. Internally, we hard code the port to the default 9090 for HTTP, and expect the user to separately override the service.externalPort value if they want it to be served externally on port 80 instead of 443 when the `enableInsecureLogin` flag is `true`.

#### Which issue this PR fixes
  - fixes #7516

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/kubernetes-dashboard/templates/NOTES.txt [hunk 1]
```diff
@@ -22,11 +30,20 @@ Get the Kubernetes Dashboard URL by running:
 
 Get the Kubernetes Dashboard URL by running:
   export SERVICE_IP=$(kubectl get svc {{ template "kubernetes-dashboard.fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+{{- if .Values.enableInsecureLogin }}
+  ech https://$SERVICE_IP/
```
Comment (desaintmartin) [inline stable/kubernetes-dashboard/templates/NOTES.txt]: Typo?

Comment (mbarrien) [reply stable/kubernetes-dashboard/templates/NOTES.txt]: Fixed

File: stable/kubernetes-dashboard/templates/NOTES.txt [hunk 2]
```diff
@@ -22,11 +30,20 @@ Get the Kubernetes Dashboard URL by running:
 
 Get the Kubernetes Dashboard URL by running:
   export SERVICE_IP=$(kubectl get svc {{ template "kubernetes-dashboard.fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+{{- if .Values.enableInsecureLogin }}
+  echo https://$SERVICE_IP/
```
Comment (desaintmartin) [inline stable/kubernetes-dashboard/templates/NOTES.txt]: http!

**Review by desaintmartin — CHANGES_REQUESTED** [review]

File: stable/kubernetes-dashboard/templates/NOTES.txt (hunk 2 above)
Comment (mbarrien) [reply stable/kubernetes-dashboard/templates/NOTES.txt]: Fixed

**Comment by ieugen** [comment]
I've manually merged the changes and they work great ! Thanks @mbarrien for the work. I hope it gets merged soon.

**Comment by tonyxiao** [comment]
This looks great! Thank you !

**Comment by desaintmartin** [comment]
Hello,
Could you rebase one last time and fix the conflicts?

**Comment by mbarrien** [comment]
@desaintmartin rebased and ready for approval

File: stable/kubernetes-dashboard/templates/NOTES.txt [hunk 3]
```diff
@@ -5,15 +5,23 @@
 {{- if .Values.ingress.enabled }}
 From outside the cluster, the server URL(s) are:
 {{- range .Values.ingress.hosts }}
+{{- if .Values.enableInsecureLogin }}
```
Comment (desaintmartin) [inline stable/kubernetes-dashboard/templates/NOTES.txt]: $.Values , as we changed our scope here.

**Comment by desaintmartin** [comment]
You also need to update the ingress to use either 443 or 80 depending on the case.

**Comment by mbarrien** [comment]
> Just tested and it requires two changes : the one from Notes I just wrote and :
> 
> You also need to update the ingress to use either 443 or 80 depending on the case. (edit: I misread, the service port does not change. Isn't it misleading that a port using 443 serves unencrypted content?)

The problem is the 443 is coming from values.yaml, which to my knowledge doesn't support templating as described in <https://github.com/helm/helm/issues/2492>. Either I change out values.yaml in a non-backwards compatible way (need some concept of a null default value with logic in 2 places to default to the right port number?), or add a new .Values.service.insecureExternalPort and add different if/else logic. Both seem ugly. Preferences?

File: stable/kubernetes-dashboard/templates/NOTES.txt (hunk 3 above)
Comment (mbarrien) [reply stable/kubernetes-dashboard/templates/NOTES.txt]: Don't understand this request. You want it to say `{{- if $.Values.enableInsecureLogin }}`? If that's the request, I've never seen such syntax before.

**Comment by desaintmartin** [comment]
I understand. Let's keep it like this and one can still change the externalPort if he wants to.

File: stable/kubernetes-dashboard/templates/NOTES.txt (hunk 3 above)
Comment (desaintmartin) [reply stable/kubernetes-dashboard/templates/NOTES.txt]: Have you tried it? ;)
$ is a magical syntax that gets the global scope. `range`changes the scope to be `.Values.ingress.hosts`, you can then access it with `.`, which is done 2 lines after. But then `.Values.something` translates within this `range` to `.Values.ingress.hosts.Values.something`.

Welcome to Helm black magic.

Comment (mbarrien) [reply stable/kubernetes-dashboard/templates/NOTES.txt]: While you were typing that, found https://github.com/helm/helm/issues/1311 which explained it. Fixed.

**Pull request merged**